internal/pkg/otel/manager/manager.go

```diff
@@ -374,6 +381,13 @@ func buildMergedConfig(
 		if err != nil {
 			return nil, fmt.Errorf("failed to generate otel config: %w", err)
 		}
+
+		// get log level from agent info
+		level := translate.GetOTelLogLevel(cfgUpdate.logLevel.String())
+		if err := componentOtelCfg.Merge(confmap.NewFromStringMap(map[string]any{"service::telemetry::logs::level": level})); err != nil {
```

Comment:
In a Hybrid Agent configuration that contains collector pipelines, this will collide with any explicitly set value of `service::telemetry::logs::level` by the user, making `agent.logging.level` take priority.

I think we want to respect `service::telemetry::logs::level` if it's explicitly set first, then use `agent.logging.level` if there is no explicit log level in the collector configuration.

Reply:
This getting handled correctly should be a test case.